### PR TITLE
feat: setup securityContext on plugins

### PR DESCRIPTION
**Pull request by nerzhul**
At Veepee we only permit pods to run as non root. Plugin works properly as non root, add variables to setup it as non root and permit to customize UID

## Description

Ref #121 
## Checklist
- [x] I've read the [guidelines for contributing](https://github.com/aquasecurity/trivy-operator/blob/main/CONTRIBUTING.md) to this repository.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy-operator/tree/main/docs) with the relevant information (if needed).
- [x] I've added usage information (if the PR introduces new options)
- [x] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by nerzhul** [comment]
I just fixed tests, it's now mergeable, if you can review and release when you can, we'd glad to do more test on our 200 nodes cluster :)

File: docs/operator/configuration.md
Comment (erikgb) [inline docs/operator/configuration.md:53]: Will only "One-line JSON representation" be supported by this? I think it would be nice to support YAML-format - if possible, as this will be a better UX IMO. But probably not possible?

File: pkg/vulnerabilityreport/builder_test.go
Comment (erikgb) [inline pkg/vulnerabilityreport/builder_test.go:213]: I don't think this test should change, if I don't misunderstand the test. As this new feature should not change the existing job spec, if not using this feature. Multiple occurrences.

File: pkg/trivyoperator/config.go
Comment (erikgb) [inline pkg/trivyoperator/config.go:59]:
```suggestion
	keyScanJobPodSecurityContext         = "scanJob.podTemplateSecurityContext"
```

To be better aligned with the existing property `scanJob.podTemplateLabels`?

File: pkg/vulnerabilityreport/builder_test.go
Comment (nerzhul) [reply pkg/vulnerabilityreport/builder_test.go:213]: ok you prefer to keep previous behaviour, and not add any empty spec, let's look at this

**Comment by erikgb** [comment]
@nerzhul Thanks for this PR! I have added a few comments. We run on Openshift, and it is really picky about securityContext. It has it's SCC that want to control all that.

File: docs/operator/configuration.md
Comment (erikgb) [reply docs/operator/configuration.md:53]: Isn't this configured from a configmap? But I see that the similar property `scanJob.templateLabel` (also a typo; should be Label**s**) is using a simple string. I'll leave this decision up to the maintainers....

Comment (nerzhul) [reply docs/operator/configuration.md]: it's the configmap key exact, i use JSON because GetScanJobTolerations uses also a JSON to be consistent

**Comment by nerzhul** [comment]
@erikgb i addressed your concerns, except the one on the format, let's wait for maintainers, but it kept consistency with tolerations, both are JSON unmarshaled.

For Openshift, you have specific things. On our side we use PSP and PodSecurities on all namespaces, then it's not possible to run root pods. With this patch pods runs perfectly as UID > 1000 users and creates reports, on a vanilla Kubernetes with just PSP/PS

**Comment by erikgb** [comment]
> For Openshift, you have specific things. 

I am not against adding this new feature, but I don't think it should break the existing operator. :-)

**Comment by nerzhul** [comment]
no problem @erikgb , now it's more close to the original one, regarding the tests on other parts :) 
Tests on our dev cluster shows it's fine.
I just amened the helm chart to make helm chart use YAML syntax and let it transform to JSON like tolerations

File: pkg/trivyoperator/config.go
Comment (erikgb) [inline pkg/trivyoperator/config.go:128]:
```suggestion
	err := json.Unmarshal([]byte(c[keyScanJobPodSecurityContext]), scanJobPodSecurityContext)
```

Comment (nerzhul) [reply pkg/trivyoperator/config.go]: fixed :) ty

**Comment by nerzhul** [comment]
Okay i hope we can merge this before you change the whole architecture :) who are maintainers ? in the changelog it seems you are 2 currently active people

**Comment by josedonizetti** [comment]
@nerzhul Thanks for the PR, didn't have time to look/test this yet, but I think your change here is addressing what was asked here? https://github.com/aquasecurity/trivy-operator/issues/121

**Comment by chen-keinan** [comment]
@nerzhul thanks you for contributing , I'm find with keeping the `podSecurityContext` in json format to be consist with this type of data.
I'll approve it later after I'll test it and make sure it works find.

**Review by chen-keinan — APPROVED** [review]

**Comment by chen-keinan** [comment]
I assume it will solve the most cases where it is required in pod level however not if it is required for a specific init container

**Review by josedonizetti — COMMENTED** [review]
Looks good! But I do have a comment/question. Considering trivy will scan and report misconfiguration for pods not having a security context with `runAsUser`,  `runAsGroup` and  `runAsNonRoot`. 

https://avd.aquasec.com/misconfig/kubernetes/general/avd-ksv-0012/
https://avd.aquasec.com/misconfig/kubernetes/general/avd-ksv-0020/
https://avd.aquasec.com/misconfig/kubernetes/general/avd-ksv-0021/

Why don't we have a default value for `podTemplateSecurityContext` considering the AVDs above?

**Review by josedonizetti — APPROVED** [review]
I'll create an issue to discuss further if we should have default or not. :)

**Comment by nerzhul** [comment]
@josedonizetti it's a very good comment, i think we should have best security options by default, i just kept compat with existing when writing this pr :)

**Comment by erikgb** [comment]
> @josedonizetti it's a very good comment, i think we should have best security options by default, i just kept compat with existing when writing this pr :)

It will be a breaking change for users on Openshift, but I do not oppose adding a default according to K8s security best practices. In this case, I think Openshift can improve. 😉 But it should be mentioned in the release notes as a breaking change IMO.

**Pull request merged**
